frontend/src/pages/CollaborationPage.js

```diff
@@ -32,28 +50,30 @@ const CollaborationPage = () => {
 
   useEffect(() => {
     if (!location.state?.room) {
-      navigate(homeUrl, { replace: true })
+      setIsPartnerLeft(true)
       return
     }
-
     collabSocket.emit('join-room', getCollabRoomId(location.state.room))
     matchingSocket.emit('join-room', location.state.room)
   }, [location.state, navigate])
 
   useEffect(() => {
     collabSocket.on('partner-disconnected', () => {
+      // TO CHECK: CHECK frontend again after matching service bug is rectified
       setIsPartnerOnline(false)
     })
     collabSocket.on('partner-connected', (roomClients) => {
-      // check if user is the only one in the room or if there are more ppl
+      // Check if user is the only one in the room or if there are more ppl
       setIsPartnerOnline(roomClients.length > 1)
     })
   }, [])
 
   useEffect(() => {
     matchingSocket.on('partner-left', (data) => {
       if (data === 'partner left') {
-        navigate(homeUrl)
+        // TO CHECK: CHECK frontend again after matching service bug is rectified
+        setIsPartnerLeft(true)
+        setIsPartnerOnline(false) // so that it will not show PartnerOfflineAlertDialog
```

Comment:
@Silvernitro due to some changes in the implementation, `isPartnerOnline` will be set to `false` so that the `PartnerOfflineAlertDialog` does not appear.